File: backend/models/media.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Media struct {
	ID uint `gorm:"primaryKey" json:"id"`

	UUID string `gorm:"size:100;uniqueIndex" json:"uuid"`

	FileName string `gorm:"size:255" json:"file_name"`

	OriginalName string `gorm:"size:255" json:"original_name"`

	FilePath string `gorm:"size:500" json:"-"`

	URL string `gorm:"size:500" json:"url"`

	MimeType string `gorm:"size:100" json:"mime_type"`

	Extension string `gorm:"size:20" json:"extension"`

	AltText string `gorm:"size:255" json:"alt_text"`

	FileSize int64 `json:"file_size"`

	Width int `json:"width"`

	Height int `json:"height"`

	CreatedAt time.Time `json:"created_at"`

	UpdatedAt time.Time `json:"updated_at"`

	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: admin/types/media.ts
export interface Media {
    id: number;
    uuid: string;
    file_name: string;
    original_name: string;
    url: string;
    mime_type: string;
    extension: string;
    alt_text?: string;
    file_size: number;
    width: number;
    height: number;
    created_at: string;
    updated_at: string;
}

export interface MediaPagination {
    page: number;
    limit: number;
    total: number;
    totalPages: number;
}

export interface MediaListResponse {
    success: boolean;
    message: string;
    data: {
        items: Media[];
        pagination: MediaPagination;
    };
}

export interface MediaResponse {
    success: boolean;
    message: string;
    data: Media;
}
